common/status-effects/weakness.ts

```diff
@@ -1,29 +1,49 @@
 import {
 	CardComponent,
 	ObserverComponent,
+	PlayerComponent,
 	StatusEffectComponent,
 } from '../components'
 import {GameModel} from '../models/game-model'
 import {beforeAttack} from '../types/priorities'
 import {Counter, statusEffect} from './status-effect'
 
-const WeaknessEffect: Counter<CardComponent> = {
+const WeaknessEffect: Counter<PlayerComponent> = {
 	...statusEffect,
 	id: 'weakness',
 	icon: 'weakness',
 	name: 'Weakness',
-	description: "This Hermit is weak to the opponent's active Hermit's type.",
+	description:
+		'[weakType] is weak to [strongType] for the duration fo this counter.',
 	counter: 3,
 	counterType: 'turns',
 	onApply(
```

Comment:
Do not apply status if the opponent's active hermit is Armor Stand. Your implementation will cause it to last forever and show the template description
```suggestion
	counterType: 'turns',
	applyCondition: (_game, value) =>
		value instanceof PlayerComponent && !!value.getActiveHermit()?.isHermit(),
	onApply(
```